pkg/metrics/producers/scheduledcapacity/schedules.go

```diff
@@ -0,0 +1,106 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduledcapacity
+
+import (
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/autoscaling/v1alpha1"
+	"github.com/robfig/cron/v3"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// parseScheduleSpecIntoCrontab returns a crontab equivalent to the strongly-typed schedule
+func parseScheduleSpecIntoCrontab(pattern *v1alpha1.Pattern) string {
```

Comment:
consider orienting these functions around a struct. Right now it's just a bag of functions, which can be difficult to organize over time and understand as a reader.

e.g.
```
type Crontab struct {
  
}

func CrontabFrom(pattern *v1alpha1.Pattern) *Crontab { ... }

func (c *Crontab) nextMatch() { ... }
```

Reply:
just strings itself is how the `cron` golang library recognizes it. I'll add a helper function to make this section clearer, though.